### ISSUE TITLE
Known issue: Contributor mismatch 

### ISSUE THREAD

**Issue by glasnt** [LABHR/octohatrack#65]
(Possible cause: Pull Requests with multiple participants)

I need to dive into this more, but it appears that the statistics for what GitHub reports in the banner of a repo and what the API endpoint is reporting might not be the same, and I think this is possibly because of issues when the participants of a Pull Request differ to the participants in a git branch

A PR would be attributed to one person, but when there's more than one person working on a branch and it gets merged, there appears to be an inconsistency. 

In the wild: https://twitter.com/nibalizer/status/696279727120654336
```
Just ran octohatrack on voxpupul/puppet-collectd:
Code Contributors: 30
Non Code Contributors: 133
```
However, https://github.com/voxpupuli/puppet-collectd reports 94 contributors

Full investigation pending

**Comment by tacaswell** [comment]
I am also seeing this on matplotlib.  octohatrack is reporting only 30 code contributors, where as scraping the git repo gives 300+ from 2013-last summer (https://gist.github.com/tacaswell/72b0d579aeb54d4fbf87).

**Comment by glasnt** [comment]
I have a sneaking suspicion that this is related to the fact that by default pagination is 30, so it's not paginating correctly. I'm currently running master code, and i'm seeing the full 94 contributors to puppet-collectd, as reported by github UI

**Comment by glasnt** [comment]
@tacaswell Which version of octohatrack are you using? The latest from pip, or the master branch? I'm not sure if it's a bug in an older version, because I ran it myself on master and got 94 contributors. 

I'm very tempted to redo the internals real soon now, because I think I've learnt enough python to be able to not just throw in vendored code into a repo

**Comment by glasnt** [comment]
I found the root cause!

The octohub code I'm using uses pagination based on the assumption that there will be a `next` in the json response that it can pull and get the next page

The `contributors` endpoint ([example](https://api.github.com/repos/voxpupuli/puppet-collectd/contributors)) doesn't have this metadata. You can get to [page 2](https://api.github.com/repos/voxpupuli/puppet-collectd/contributors?page=2) but the link to this isn't contained within the response. 

Thus, only the first page is traversable via the helper code. 

I'll consolidate the functionality I require, and shop around for a helperlib that handles all the cases I require. If I can't, I'll hack about with my copy of octohub to workaround this particular issue.

**Comment by glasnt** [comment]
Initial fix in #76. If the API changes, this code needs to be updated.
